src/util/translation.h

```diff
@@ -6,10 +6,43 @@
 #define BITCOIN_UTIL_TRANSLATION_H
 
 #include <tinyformat.h>
+#include <util/string.h>
 
+#include <cassert>
 #include <functional>
 #include <string>
 
+/** Translate a message to the native language of the user. */
+const extern std::function<std::string(const char*)> G_TRANSLATION_FUN;
+
+namespace util {
+/**
+ * Translation function.
+ * If no translation function is set, simply return the input.
+ */
+inline std::string translate(const char* lit)
+{
+    return G_TRANSLATION_FUN ? G_TRANSLATION_FUN(lit) : lit;
+}
+/** Type to denote whether an orginal string literal is translatable */
+template <bool translatable = true>
```

Comment:
nit: I would prefer templating this on an enum `{Translateable,Untranslateable}` for readability over the non-obvious `<{true,false}>` boolean

Reply:
> nit: I would prefer templating this on an enum `{Translateable,Untranslateable}` for readability over the non-obvious `<{true,false}>` boolean

Note: with changes in suggested branch it should no longer be necessary for this to be templated.